src/app/charts/chart-base/chart-base.component.ts

```diff
@@ -191,26 +207,53 @@ export class ChartBaseComponent implements AfterViewInit, OnDestroy {
     this.brush = d3
       .brushX()
       .extent([[0, 0], [this.width, this.height]])
-      .on('end', () => this.brushed(this.xScale))
+      .on('end', () => this.brushed())
+
+    if (this.sensorDataTimeFrame) {
+      this.updateBrushFromSensorDates()
+    }
 
     this.chart
       .append('g')
       .attr('class', 'brush')
       .call(this.brush)
+      .transition()
+      .duration(1000)
       .call(
         this.brush.move,
-        this.brushExtent
-          ? this.brushExtent
-          : [this.width - this.brushWidthDefault, this.width]
+        !this.brushExtentFromSensors
+          ? [this.width - this.brushWidthDefault, this.width]
+          : this.brushExtentFromSensors
       )
   }
 
-  private brushed(xScale) {
-    this.brushExtent = d3.event.selection
+  private brushed() {
+    if (!this.brushUpdatedFromSensors) {
+      this.brushExtent = d3.event.selection
+
+      this.brushMove.emit([
+        this.xScale.invert(this.brushExtent[0]),
+        this.xScale.invert(this.brushExtent[1])
+      ])
+    } else {
+      this.brushUpdatedFromSensors = false
+      this.brushMove.emit(this.sensorDataTimeFrame)
+    }
+  }
 
-    this.brushMove.emit([
-      xScale.invert(this.brushExtent[0]),
-      xScale.invert(this.brushExtent[1])
-    ])
+  private updateBrushFromSensorDates() {
+    const brushLeft = this.xScale(this.sensorDataTimeFrame[0])
```

Comment:
Minor comment. this would be better understood if called `brushTimeStart` `brushTimeEnd`.